microsoft/botframework-solutions
# calendar bugs fix

## Description
fix some bugs

## Related Issue
1. state won't clear when cancel the flow
2. the flow went wrong when input "create a calendar appointment at 3:30 tomorrow for half an hour"
3. flow will skip the update step when input wrong name twice
4. the recognized entities will be overwritten by following input
5. won't read the detail of meeting when choose a meeting in summary flow
6. the email address can't be recognized when retry to input attendee name
7. can't find meetings by a specific time in update and delete flow
8. response "yes" will cancel the flow when bot asked "Do you want to hear more? Which one do you want to hear?"
9. add user name in person not found prompt

## Testing Steps
1. input "create a meeting with yue" -> "cancel" -> "yes" -> "create a meeting"
the flow should ask the attendees again
2. input "create a calendar appointment at 3:30 tomorrow for half an hour"
the duration should be recognized
3. input "create a meeting" -> "skgjlej(the name that can't be find)" -> "skgjlej"
bot will ask for name again until cancel or get a correct name
4. start a create meeting flow, when ask for content, say "team plan"(will be recognized as title). And bot won't use "team plan" to overwrite the real title.
5. input "what are my meetings today" -> "first one"
listen to the speech. will read out the meeting title and time
6. input "create a meeting" -> "skgjlej(the name that can't be find)" -> "yuesu@microsoft.com"
bot will save this address as attendee
7. input "update my meeting" (or input "cancel my meeting") -> "6pm"(the time you have meeting)
the meeting should be found
8. input "what are my meetings today" -> "yes"
should ask user to input number
9. input "create a meeting" -> "skgjlej(the name that can't be find)"
bot will say the can't find wording with the name
